src/azure-cli-core/azure/cli/core/application.py

```diff
@@ -154,13 +154,14 @@ def execute(self, unexpanded_argv):  # pylint: disable=too-many-statements
 
         # Rudimentary parsing to get the command
         nouns = []
-        for noun in argv:
+        for i in range(len(argv)):  # pylint: disable=consider-using-enumerate
             try:
-                if noun[0] == '-':
+                if argv[i][0] == '-':
```

Comment:
`noun.lower()` creates a lowercase copy of the argv component without actually changing it.  If I try to parse the command line without modifying the argv component, it raises a ParseError.